discovery/gossiper.go

```diff
@@ -2579,6 +2672,31 @@ func (d *AuthenticatedGossiper) handleChanAnnouncement(nMsg *networkMsg,
 			_, _ = d.recentRejects.Put(key, &cachedReject{})
 		}
 
+		if !nMsg.isRemote {
```

Comment:
Q: I wonder when this could happen that we locally cannot add an edge ? Should we maybe log this case ?

Reply:
Can log it, it could happen if we introduce a bug or if a database transaction fails. We need the check here so that we don't panic below with `nMsg.peer.PubKey()` since `peer` isn't set for local announcements.